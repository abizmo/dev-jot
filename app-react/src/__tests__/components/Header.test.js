import { shallow } from 'enzyme';

import Header from '../../components/Header';

describe('Header Tests', () => {
  test('should render without profile and match snapshot', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h1').text()).toBe('dev-jot');
    expect(wrapper.find('.profile')).toHaveLength(0);
  });

  test('should render with profile and match snapshot', () => {
    const wrapper = shallow(<Header profileVisible avatarImgPath="#" />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.avatar').children().prop('src')).toBe('#');
  });
});
