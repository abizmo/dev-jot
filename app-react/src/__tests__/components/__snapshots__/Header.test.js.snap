// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Tests should render with profile and match snapshot 1`] = `
<styled.header
  className="header"
>
  <styled.div
    className="header-title"
  >
    <h1>
      dev-jot
    </h1>
    <img
      alt="note icon"
      src="sticky_note-white.svg"
    />
  </styled.div>
  <styled.div
    className="profile"
  >
    <div
      className="avatar"
    >
      <styled.img
        alt="avatar"
        referrerpolicy="no-referrer"
        src="#"
      />
    </div>
  </styled.div>
</styled.header>
`;

exports[`Header Tests should render without profile and match snapshot 1`] = `
<styled.header
  className="header"
>
  <styled.div
    className="header-title"
  >
    <h1>
      dev-jot
    </h1>
    <img
      alt="note icon"
      src="sticky_note-white.svg"
    />
  </styled.div>
</styled.header>
`;
